src/tables.jl

```diff
@@ -207,16 +239,166 @@ function _colnames(s::AbstractDimStack)
 
 
+# WideDimTable
+
+
+"""
+    WideDimTable <: AbstractDimTable
+
+    WideDimTable(A::AbstractDimArray)
+
+Construct a Tables.jl/TableTraits.jl compatible object out of an `AbstractDimArray`.
+
+This table will have a column for the array data and columns for each
+`Dimension` index, as a [`DimColumn`]. These are lazy, and generated
+as required.
+
+Column names are converted from the dimension types using
+[`DimensionalData.dim2key`](@ref). This means type `Ti` becomes the
+column name `:Ti`, and `Dim{:custom}` becomes `:custom`.
+
+To get dimension columns, you can index with `Dimension` (`X()`) or
+`Dimension` type (`X`) as well as the regular `Int` or `Symbol`.
+"""
+struct WideDimTable{DS} <: AbstractDimTable
+    colnames::Vector{Symbol}
+    dimcolumns::DS
+    dimarraycolumns::Vector{DimArrayColumn}
+end
+
+function WideDimTable(s::AbstractDimStack; mergedims=false)
+    dims_ = dims(s)
+    dimcolumns = collect(map(d -> DimColumn(d, dims_), dims_))
+    dimarraycolumns = collect(map(A -> DimArrayColumn(A, dims_), s))
+
+    if mergedims
+        dimcol = MergedDimColumn(Tuple(dimcolumns), :geometry)
+        keys = vcat([:geometry], collect(_colnames(s))[length(dims_)+1:end])
+        return WideDimTable(keys, [dimcol], dimarraycolumns)
+    else
+        keys = collect(_colnames(s))
+        return WideDimTable(keys, dimcolumns, dimarraycolumns)
+    end
+end
+
+function WideDimTable(xs::Vararg{AbstractDimArray}; layernames=[Symbol("layer_$i") for i in eachindex(xs)], mergedims=false)
+    # Construct DimColumns
+    dims_ = dims(first(xs))
+    dimcolumns = map(d -> DimColumn(d, dims_), dims_)
+    dimnames = collect(map(dim2key, dims_))
+
+    # Construct DimArrayColumns
+    dimarraycolumns = collect(map(A -> DimArrayColumn(A, dims_), xs))
+
+    # Merge DimColumns
+    if mergedims
+        colnames = vcat([:geometry], layernames)
+        dimcol = MergedDimColumn(Tuple(dimcolumns), :geometry)
+        return WideDimTable{typeof(dimcol)}(colnames, dimcol, dimarraycolumns)
+    else
+        colnames = vcat(dimnames, layernames)
+        return WideDimTable{typeof(dimcolumns)}(colnames, dimcolumns, dimarraycolumns)
+    end
+end
+
+function WideDimTable(x::AbstractDimArray; layersfrom=nothing, mergedims=false)
+    if !isnothing(layersfrom) && (layersfrom <: Dimension) && (any(isa.(dims(x), layersfrom)))
+        nlayers = size(x, layersfrom)
+        layers = [(@view x[layersfrom(i)]) for i in 1:nlayers]
+        layernames = Symbol.(["$(dim2key(layersfrom))_$i" for i in 1:nlayers])
+        return WideDimTable(layers..., layernames=layernames, mergedims=mergedims)
+    else
+        # Construct DimColumns
+        dims_ = dims(x)
+        dimcolumns = map(d -> DimColumn(d, dims_), dims_)
+        dimnames = collect(map(dim2key, dims_))
+
+        # Construct DimArrayColumn
+        dimarraycolumn = DimArrayColumn(x, dims_)
+
+        # Merge DimColumns
+        if mergedims
+            colnames = vcat([:geometry], [:value])
```

Comment:
Why `:value` here?

Reply:
When converting an `AbstractDimArray` to a table, we need to come up with a name for the one and only `DimArrayColumn`. `:value` seemed like a good choice.

Reply:
First we can try to use `Symbol(name(dimarray)))`, but check its not `Symbol("")` and it is then `:value` is probably as good as any